### PR TITLE
Fixes ledgerwatch/erigon#2421

### PR DESCRIPTION
**Pull request by Cript**

### CODE REVIEW

**Comment by Cript** [comment]
https://github.com/ledgerwatch/erigon-lib/pull/72 should be merged before this

File: ethdb/prune/storage_mode.go [hunk 1]
```diff
@@ -137,6 +154,9 @@ type Mode struct {
 type Distance uint64
 
 func (p Distance) Enabled() bool { return p != math.MaxUint64 }
+func (p Distance) toValue() uint64 { return uint64(p) }
+func (p Distance) useDefaultValue() bool { return uint64(p) == params.FullImmutabilityThreshold }
+func (p Distance) dbType() []byte { return kv.PruneHistoryTypeOlder }
```
Comment (AskAlexSharov) [inline ethdb/prune/storage_mode.go:159]: probably need rename `kv.PruneHistoryTypeOlder` to `kv.PruneTypeOlder`, because "history" word reserved for "historical state access" meaning.

File: ethdb/prune/storage_mode.go [hunk 2]
```diff
@@ -148,38 +168,54 @@ func (p Distance) PruneTo(stageHead uint64) uint64 {
...
 
+// Before number after which keep in DB
+type Before uint64
+
+func (b Before) Enabled() bool { return b > 0 }
+func (b Before) toValue() uint64 { return uint64(b) }
+func (b Before) useDefaultValue() bool { return uint64(b) == 0 }
+func (b Before) dbType() []byte { return kv.PruneHistoryTypeBefore }
+
+func (b Before) PruneTo(uint64) uint64 {
```
Comment (AskAlexSharov) [inline ethdb/prune/storage_mode.go:179]: please add tests for PruneTo method. Because it's very dangerous functionality - can lead to whole db drop. Need test that it handles `uint underflow`, `uint overflow`.

Comment (Cript) [reply ethdb/prune/storage_mode.go:179]: Changes in https://github.com/ledgerwatch/erigon-lib/pull/76

File: ethdb/prune/storage_mode.go (hunk 1 above)
Comment (Cript) [reply ethdb/prune/storage_mode.go]: Changes in https://github.com/ledgerwatch/erigon-lib/pull/76

File: ethdb/prune/storage_mode.go (hunk 2 above)
Comment (Cript) [reply ethdb/prune/storage_mode.go:179]: Tests are added. Also I would like to add tests with `before` in `eth/stagedsync/stage_execute_test.go` 
And maybe also in `turbo/stages/blockchain_test.go`?
